core/src/main/scala/chisel3/internal/firrtl/Converter.scala

```diff
@@ -219,7 +219,11 @@ private[chisel3] object Converter {
           convert(probe, ctx, sourceInfo)
         )
       )
-    case e @ Verification(_, op, info, clk, pred, msg) =>
+    case e @ Verification(_, op, info, clk, pred, pable) =>
+      val (fmt, args) = unpack(pable, ctx)
+      if (op == Formal.Cover && args.nonEmpty) {
+        throwException("cover message cannot be used as a format string")
+      }
```

Comment:
This can be done as a `require` inside the constructor of the `Verification` op. (Arguably that motivates splitting this into separate IR structures for assert, assume, and cover, but that's not necessary.)